tests/async/test_async_auth_decorators.py

```diff
@@ -0,0 +1,145 @@
+from functools import partial
+
+from django.conf import settings
+from django.contrib.auth import models
+from django.contrib.auth.decorators import login_required, permission_required
+from django.core.exceptions import PermissionDenied
+from django.http import HttpResponse
+from django.test import TestCase
+from django.test.client import RequestFactory
+
+
+class AsyncLoginRequiredTestCase(TestCase):
+    """
+    Tests the login_required decorators with async views.
+    """
+
+    factory = RequestFactory()
+
+    @staticmethod
+    async def auser(user):
+        return user

```

Comment:
Can we change it to the class method and always return `cls.user`?

Reply:
I think this could be a little difficult, because `request.auser` doesn't always return `cls.user`, it sometimes returns an `AnonymousUser`, such as https://github.com/django/django/pull/17621/commits/c73ee01b7af97106d6bd7d40f9a036f395f979c3#diff-c14ba26ed5b9575e08fdb8baf770ff7cd0d8d999db0c325345b317615e9d9784R101

But the same method in the `PermissionsRequiredDecoratorTest` can be a classmethod. How do you think about?